girder/__init__.py

```diff
@@ -259,6 +260,26 @@ def logprint(*args, **kwargs):
         _originalStdOut.flush()
 
 
+def _setupCache():
+    """
+    Setup caching based on configuration file.
+
+    If the cache is disabled, both the global and per-request cache are set
+    to the null backend, effectively making them a noop. This allows for
+    decorating functions with the cache region, despite it being unused.
+    """
+    curConfig = config.getConfig()
+
+    if not curConfig['cache']['enabled']:
+        cache.configure(backend='dogpile.cache.null')
+        requestCache.configure(backend='dogpile.cache.null')
+    else:
+        cache.configure_from_config(curConfig['cache'], 'cache.global.')
+        requestCache.configure_from_config(curConfig['cache'], 'cache.request.')
+
+
+_setupCache()
```

Comment:
Is `girder.__init__` the right place to run this? Or would it be better to have it run inside the server configuration or something? I'm trying to think of whether users of the girder python library for things other than girder-server might want to use it without the need for caching.

Reply:
> Is girder.__init__ the right place to run this?

I don't think it is. The more I work with Python the more I find myself having a strong dislike for import side effects.